### ISSUE TITLE
Drag and drop of an element between top window and iframe is not working. 

### ISSUE THREAD

**Issue by chowdhur** [SeleniumHQ/selenium#7999]
## 🐛 Bug Report
Drag and drop of an element between top window and iframe is not working. 

## To Reproduce
Unable to drag an element from top window and drop on another element present within an iframe. Mouse is holding that draggable element but not dropping on the droppable element. Below I mention OS and Browser information.

## Environment
OS: MacOS Catalina
Browser: Safari 
Browser version: 13.0.3 (15608.3.10.1.4)
Browser Driver version: selenium-safari-driver-3.141.59.jar
Selenium version: 3.141.59

Let me know if you need more details.

**Comment by sachinboob** [comment]
@chowdhur  Could you please also add the code that you are using to achieve this? If you are not able to share the application that you trying to automate it is fine.

**Comment by chowdhur** [comment]
Hi Sachin,

Thanks for addressing this issue!
For my case, image to drag is on top window, image to drop is on iframe which is within top window.
I tried below mechanisms for this:

1. Switching iframes between drag and drop:
----------------------------------------------
act.clickAndHold(imageToDrag).build().perform();  
switchToFrame("//iframe[@id='cq-cf-frame']");
act.moveToElement(imageDropField).release(imageDropField).build().perform();

2. Another try:
---------------
WebElement imageToDrag = locateElement("(//div[@class='thumb-wrap'])[1]//div");
Actions builder = new Actions(globalWebDriver);          
Actions action = builder.clickAndHold(imageToDrag);
action.build().perform();
switchToiFrame("//iframe[@id='cq-cf-frame']");  
WebElement imageDropField = locateElement("//span[text()='Image']/parent::div/following-sibling::div//img[@class='cq-image-placeholder']");
builder.moveToElement(imageDropField).build().perform();
//Release mouse at destination 
builder.release(imageDropField).build().perform();

3. Tried using moveByOffset method:
---------------------------------------
int dropLocationX = imageDropField.getLocation().getX();
int dropLocationY = imageDropField.getLocation().getY();*/
globalWebDriver.switchTo().defaultContent();
int dragLocationX = imageToDrag.getLocation().getX();
int dragLocationY = imageToDrag.getLocation().getY();

act.moveToElement(imageToDrag).clickAndHold().moveByOffset(dropLocationX-dragLocationX, dropLocationY-dragLocationY).release(imageToDrag).build().perform();

Thanks,
Rajdeep
